File: Daily Challenge/C/Complete The Row - Pascal's Triangle 15.9.2025.c
#include <stdio.h>

void generate_row(long long row[], int r) {
    row[0] = 1;
    for (int k = 1; k <= r; k++)
        row[k] = row[k - 1] * (r - k + 1) / k;
}

int main() {
    int n;
    scanf("%d", &n);
    long long nums[n];
    for (int i = 0; i < n; i++) scanf("%lld", &nums[i]);

    long long row[100];
    for (int r = n - 1; r < 50; r++) {
        generate_row(row, r);
        int i = 0;
        for (int j = 0; j <= r; j++) {
            if (i < n && row[j] == nums[i]) i++;
        }
        if (i == n) {
            for (int j = 0; j <= r; j++)
                printf("%lld%c", row[j], j == r ? '\n' : ' ');
            break;
        }
    }
    return 0;
}

File: Daily Challenge/Python/Complete The Row - Pascal's Triangle 15.9.2025.py
n = int(input().strip())
nums = list(map(int, input().split()))

def generate_row(r):
    row = [1]
    for k in range(1, r + 1):
        val = row[-1] * (r - k + 1) // k
        row.append(val)
    return row

for r in range(n - 1, 50):
    row = generate_row(r)
    i = 0
    for val in row:
        if i < n and val == nums[i]:
            i += 1
    if i == n:
        print(' '.join(map(str, row)))
        break
